src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -237,7 +237,10 @@ public static MetadataReference MscorlibRef
             {
                 if (s_mscorlibRef == null)
                 {
-                    s_mscorlibRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_mscorlibRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll"),
+                        null);
```

Comment:
Looks like `SystemCoreRef` uses `CompareExchange` as well.

Besides reference equality, a helper would also simplify the property getters.
```
public static MetadataReference SystemDataRef => 
    GetReference(ref s_systemDataRef, AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Data).GetReference(display: "System.Data.v4_0_30319.dll"));